code/go/internal/validator/semantic/format_checkers.go

```diff
@@ -37,10 +42,18 @@ func (r relativePathChecker) IsFormat(input interface{}) bool {
 	}
 
 	path := filepath.Join(r.currentPath, asString)
-	_, err := os.Stat(path)
+	info, err := os.Stat(path)
 	if err != nil {
 		return false
 	}
+
+	// TODO: It happens that we want the same max size for all the files we reference with
+	// relative paths, but it'd be better if we could find a way to parameterize format
+	// checkers so we can configure specific max sizes, and we can provide better feedback.
```

Comment:
Library we use for jsonschema can only be customized with format checkers, what is not enough to add further attributes that could be passed as parameters to formatters and allow schemas like this one:
```
          src:
            description: Relative path to the screenshot's image file.
            type: string
            format: relative-path
            maxSize: 3MB
            examples:
              - "/img/apache_httpd_server_status.png"
```
Or even better, define new types, so we could define schemas like this one:
```
          src:
            description: Relative path to the screenshot's image file.
            type: relative-path
            maxSize: 3MB
            examples:
              - "/img/apache_httpd_server_status.png"
```

I wonder if we should consider a more customizable library, such as these ones:
* https://pkg.go.dev/github.com/qri-io/jsonschema
* https://pkg.go.dev/github.com/santhosh-tekuri/jsonschema

Reply:
If so, then let's treat it as a separate issue. I remember that the initial idea was to have an option to deliver a spec parser for different languages (MVP: Go, JS), but we failed to build it.

Reply:
Yes, this is not needed by now, but only because it happens that all the relative paths we need to check have the same limits.